src/frontend/components/InstructorDashboardVOD/widgets/UploadTimedTextTracksBaseWidgetGeneric/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { Box, Button } from 'grommet';
+import React, { useEffect, useRef, useState } from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+
+import { DashboardVideoLiveItemList } from 'components/DashboardVideoLiveControlPane/customs/DashboardVideoLiveItemList';
+import { useUploadManager } from 'components/UploadManager';
+import { createTimedTextTrack } from 'data/sideEffects/createTimedTextTrack';
+import { useTimedTextTrack } from 'data/stores/useTimedTextTrack';
+import { useTimedTextTrackLanguageChoices } from 'data/stores/useTimedTextTrackLanguageChoices';
+import { LanguageChoice } from 'types/LanguageChoice';
+import { modelName } from 'types/models';
+import { timedTextMode, uploadState } from 'types/tracks';
+import { Nullable } from 'utils/types';
+import { LanguageSelect } from './LanguageSelect';
+import { TimedTextTrackItem } from './TimedTextTrackItem';
+
+const messages = defineMessages({
+  noUploadFilesMessage: {
+    defaultMessage: 'No uploaded files',
+    description: 'Message displayed when there is no uploaded document.',
+    id: 'components.UploadTimedTextTracksBaseWidgetGeneric.noUploadFilesMessage',
+  },
+  uploadButtonLabel: {
+    defaultMessage: 'Upload file',
+    description: 'Label of the button used to upload the timed text file.',
+    id: 'components.UploadTimedTextTracksBaseWidgetGeneric.uploadButtonLabel',
+  },
+});
+
+interface UploadTimedTextTracksBaseWidgetProps {
+  timedTextModeWidget: timedTextMode;
+}
+
+export const UploadTimedTextTracksBaseWidgetGeneric = ({
```

Comment:
```suggestion
export const BaseUploadTimedTextTracks = ({
```

I think it's enough :wink:

Reply:
ok, I will do as you suggest, the shorter the better haha